accounts/abi/bind/precompiletemplate.go

```diff
@@ -0,0 +1,312 @@
+// These areas are highlighted with comments "ADD YOUR PRECOMPILE HERE".
+// For testing take a look at other precompile tests in core/stateful_precompile_test.go
+
+/* General guidelines for precompile development:
+1- Read the comment and set a suitable contract address in precompile/params.go. E.g:
+	{{.Contract.Type}}Address = common.HexToAddress("ASUITABLEHEXADDRESS")
+2- Set gas costs here
+3- Unless you know what you're doing write your custom code only after highlighted areas with "CUSTOM CODE STARTS HERE".
+Typically, custom codes are required in only those areas.
+4- Add your precompile upgrade in params/config.go
+5- Add your upgradable config in params/precompile_config.go
+6- Add your solidity interface and test contract to contract-examples/contracts
+7- Write solidity tests for your precompile in contract-examples/test
+8- Create e2e test for your solidity test in tests/e2e/solidity/suites.go
+9- Create your genesis with your precompile enabled in tests/e2e/genesis/
+10- Run your e2e precompile Solidity tests with 'E2E=true ./scripts/run.sh'
+
+*/
+
+package precompile
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	{{- range .Contract.Funcs}}
+	{{.Normalized.Name}}GasCost uint64 = -1 // SET A GAS COST HERE
+	{{- end}}
+
+	// {{.Contract.Type}}RawABI contains the raw ABI of {{.Contract.Type}} contract.
+	{{.Contract.Type}}RawABI = "{{.Contract.InputABI}}"
+)
+
+// Singleton StatefulPrecompiledContract and signatures.
+var (
+	_ StatefulPrecompileConfig = &{{.Contract.Type}}Config{}
+
+	{{.Contract.Type}}Precompile StatefulPrecompiledContract = create{{.Contract.Type}}Precompile({{.Contract.Type}}Address)
+	{{- range .Contract.Funcs}}
+	{{decapitalise .Normalized.Name}}Signature = CalculateFunctionSelector("{{.Original.Sig}}")
+	{{- if not .Original.IsConstant}}
+
+	ErrCannot{{.Normalized.Name}} = errors.New("non-enabled cannot {{.Original.Name}}")
+	{{- end}}
+	{{- end}}
+
+	{{.Contract.Type}}ABI abi.ABI // will be filled by init func
+)
+
+// {{.Contract.Type}}Config {{if .Contract.AllowList}}wraps [AllowListConfig] and uses it to implement {{else}}implements{{end}} the StatefulPrecompileConfig
+// interface while adding in the {{.Contract.Type}} specific precompile address.
+type {{.Contract.Type}}Config struct {
+	{{- if .Contract.AllowList}}
+	AllowListConfig
+	{{- end}}
+	UpgradeableConfig
+}
+
+{{$structs := .Structs}}
+{{range $structs}}
+	// {{.Name}} is an auto generated low-level Go binding around an user-defined struct.
+	type {{.Name}} struct {
+	{{range $field := .Fields}}
+	{{$field.Name}} {{$field.Type}}{{end}}
+	}
+{{- end}}
+
+{{- range .Contract.Funcs}}
+{{ if len .Normalized.Inputs | lt 1}}
+type {{capitalise .Normalized.Name}}Input struct{
+{{range .Normalized.Inputs}} {{capitalise .Name}} {{bindtype .Type $structs}}; {{end}}
```

Comment:
nit: seems `;` is not necessary?

Reply:
I think this is necessary for the golang, otherwise it will keep adding fields to each other in the same line. like:
` Addr common.Address  Amount *big.Int ` but `;` can separate those and gofmt can just create a new line by deleting `;`